### ISSUE TITLE
Shouldn't TypedArray<T>.wrap return TypedArray<T>

### ISSUE THREAD

**Issue by sampaioletti** [AssemblyScript/assemblyscript#938]
https://github.com/AssemblyScript/assemblyscript/blob/ce0457e9edd87c04179ce16ee99753205521aa21/std/assembly/index.d.ts#L1224

@MaxGraey looks like you changed this to return a ArrayBufferView in #865. So you dont get intellisense if you wrap a ArrayBuffer in a typed array unless you cast/assert it back.

```typescript
let buf=new ArrayBuffer(32)
let arr=Uint8Array.wrap(buf) //<-type is ArrayBufferView
let l=arr.length //<-fails
let l=(arr as Uint8Array).length //<-works
```
I was going to PR but I figured i should ask if there was a reason first.

**Comment by jtenner** [comment]
Yeah shouldn't this be `TypedArray.wrap<T extends ArrayBufferView>(buffer: ArrayBuffer, byteOffset?: i32, length?: i32): T`

**Comment by MaxGraey** [comment]
@jtenner it [still problematic](https://www.typescriptlang.org/play/?ssl=11&ssc=11&pln=11&pc=22#code/CYUwxgNghgTiAEUBGBnALjKY30lFK8AKgJ4AOIwAgjJiQDxEB88AlgLZkQjsgB2aQjToAhAK4AzCSBgA1ViADu8AN4AoeJvhJJ0mAC54wqCXFSZGrUhJoQAGX4BzNAAtDfMeyQyA3Jc3WtgDyUiggaO6e3jB+-vDoUGisYPCKmGT0AKrwIAAetnzAQrQmZnrySkwAFK6sKIaZADTaujKGxqatMACUDX4AvmpqeATwmawCABwdAPo5+fxFxOSUHfQeXjIsKvCDwwD2fOjwOAC8YxNo0yUkMwB0aVBkVXxKRjdlMlUATN3dPkA)

**Comment by MaxGraey** [comment]
But we could try [something like this](https://www.typescriptlang.org/play/?ssl=6&ssc=69&pln=6&pc=52#code/CYUwxgNghgTiAEUBGBnALjKY30lFK8AKgJ4AOIwAgjJiQDxEB88AlgLZkQjsgB2aQjToAhAK4AzCSBgA1ViADu8AN4AoeJvhJJ0mAC54wqCXFSZGrUhJoQAGX4BzNAAtDfMeyQyA3Jc3WtgDyUiggaO6e3jB+-vDoUGisYPCKmGT0AKrwIAAetnzAhGjkIAD2EsSl1LQmTAAUrqwohpkANNq6MobGpl0wAJStfgC+amp4BPCZrAIAHL0A+jn5-EVVFDV09B5eMiwq8GMTZXzo8DgAvNOzaAu1JIsAdGlQZPV8SkYPZnr1AEwDAY+IA)

**Comment by MaxGraey** [comment]
Ok, [final solution](https://www.typescriptlang.org/play/#code/CYUwxgNghgTiAEkoGdnwCoE8AOJgEEYYpMAedAPngEsBbbCEWkAOwBc1DjMAhAVwBmAkDABq1EAHd4AbwBQ8RfABGg4TABc8LiX5CRCpcsxsQAGVYBzNgAstLPrWUiA3IcXHTAeSHIQbe0dnGDd3eAEAewjAp1c5MOQ2KDZqMHhJYmxSAFV4EAAPUxZgNDYcEAiBDHKCIhIAGngASXgAXmaWRKgWMBAsXByKCgAKW2pkLWzG1X1NbTreNREASi0mtwBfeKRUeGzqdgAOHUwAfTzC1hLq3FruUgdYmCp5LbkwCM62eG-2-aOTqcAHQZKDYYYsKTzbh6dTDABMy2WbjYQJm6hRQMiERcQA)

But that's little complicated. May be somebody know better?

**Comment by MaxGraey** [comment]
Fixed in #939
